### ISSUE TITLE
Update issuance tables to align with data model change

### ISSUE THREAD

**Issue by technicallyty** [regen-network/regen-ledger#1075]
## Summary

Our data model for ecocredits has changed, however we still use ClassIssuer table to check if a `project` can be created by certain accounts under a given credit class. This table name doesn't seem to align with what is actually being checked.  

### Questions to Consider

- should we change the name of the `ClassIssuer` table? 
- should we add a `ProjectIssuer` table approving certain accounts to issue batches from a given project?

____

## For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by ryanchristo** [comment]
I don't think we want to change anything here. `ClassIssuer` is a join table used to store the credit class issuers for each credit class. Any credit class issuer should be able to issue credits for any project under the credit class.

We check `ClassIssuer` when creating a project because in the current implementation we take a permissioned approach to project creation. Until we implement functionality for a project to apply to a credit class and to be accepted to a credit class, only an approved credit class issuer should be able to create a project for the credit class.

We are planning to make project creation permissionless in the future (#882) alongside support for multiple credit classes per project (#883). These two issuers are related because the goal would be to have any address create a project, starting with no credit classes, and then the project admin would be able to apply to a credit class and the credit class admin would be able to approve the project. When any address can create a project, we would no longer check `ClassIssuer`.

**Comment by technicallyty** [comment]
> Any credit class issuer should be able to issue credits for any project under the credit class.

this is the part im tripped up on. doesn't that mean bob from project C01-001 can issue batches from project C01-002? shouldn't he have to be a part of that project to do so? unless thats just an oversight/bug/missed in the current impl 🤷🏻 (currently we only check that they're in the `ClassIssuer` list).

**Comment by ryanchristo** [comment]
> doesn't that mean bob from project C01-001 can issue batches from project C01-002?

Who is bob? Is bob a credit class issuer, a project admin, or both a credit class issuer and a project admin? If bob is a credit class issuer for credit class `C01`, he should be able to issue credit batches for project `C01-001` and `C01-002`. If bob is the admin of project `C01-001`, he should be able to update project `C01-001`. If bob is both the credit class issuer for credit class `C01` and the admin for project `C01-001`, he should be able to issue credit batches for project `C01-001` and `C01-002` and update project `C01-001`. If bob was a credit class issuer for credit class `C01` at the time of creating project `C01-001` but is no longer a credit class issuer for credit class `C01`, he should not be able to issue credit batches for project `C01-001` or `C01-002`.

**Comment by ryanchristo** [comment]
> shouldn't he have to be a part of that project to do so? unless thats just an oversight/bug/missed in the current impl 🤷🏻 (currently we only check that they're in the `ClassIssuer` list during `CreateBatch`).

Whether or not bob is the admin of a project does not determine whether or not bob can issue credit batches. Only credit class issuers can issue credit batches. We should be checking `ClassIssuer` at the time of creating a project and again at the time of creating a credit batch. Only credit class issuers can create credit batches and only credit class issuers can create projects (until we provide functionality that allows a project to apply to and be accepted to a credit class and to be created without being directly linked to a credit class). Only a project admin can update a project. A credit class issuer becomes the project admin upon creation and then may choose to update the project admin to another address that is not a credit class issuer (i.e. the project manager).

**Comment by ryanchristo** [comment]
Closing but feel free to reopen if you think there is still an issue here.
